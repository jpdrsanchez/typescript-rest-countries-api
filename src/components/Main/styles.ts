import styled from 'styled-components'

export const Wrapper = styled.main`
  color: #171923;
  min-height: 100vh;
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;
  gap: 1.25rem;

  h1 {
    font-size: 2rem;
    font-weight: 700;
    text-transform: uppercase;
  }
`
