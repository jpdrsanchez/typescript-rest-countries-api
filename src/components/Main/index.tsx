import * as S from './styles'

const Main = () => {
  return (
    <S.Wrapper>
      <h1>Genesis Boilerplate</h1>
      <p>NextJS Boilerplate for my personal and comercial projects</p>
    </S.Wrapper>
  )
}

export default Main
