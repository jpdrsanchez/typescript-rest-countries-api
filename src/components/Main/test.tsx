import { render, screen } from '@testing-library/react'

import Main from '.'

describe('<Main />', () => {
  it('Renders the heading', () => {
    const { container } = render(<Main />)

    expect(
      screen.getByRole('heading', { name: /genesis boilerplate/i })
    ).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })

  it('Renders the paragraph', () => {
    const { container } = render(<Main />)

    expect(
      screen.getByText(
        /nextJS boilerplate for my personal and comercial projects/i
      )
    ).toBeInTheDocument()

    expect(container.lastChild).toMatchSnapshot()
  })

  it('Renders the colors correctly', () => {
    const { container } = render(<Main />)

    expect(container.firstChild).toHaveStyle({ color: '#171923' })
  })
})
