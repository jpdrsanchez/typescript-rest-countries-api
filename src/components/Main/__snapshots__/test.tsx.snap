// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> Renders the heading 1`] = `
.c0 {
  color: #171923;
  min-height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 1.25rem;
}

.c0 h1 {
  font-size: 2rem;
  font-weight: 700;
  text-transform: uppercase;
}

<main
  class="c0"
>
  <h1>
    Genesis Boilerplate
  </h1>
  <p>
    NextJS Boilerplate for my personal and comercial projects
  </p>
</main>
`

exports[`<Main /> Renders the paragraph 1`] = `
.c0 {
  color: #171923;
  min-height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 1.25rem;
}

.c0 h1 {
  font-size: 2rem;
  font-weight: 700;
  text-transform: uppercase;
}

<main
  class="c0"
>
  <h1>
    Genesis Boilerplate
  </h1>
  <p>
    NextJS Boilerplate for my personal and comercial projects
  </p>
</main>
`
